src/test/java/com/configcat/ConfigV2EvaluationTest.java

```diff
@@ -0,0 +1,204 @@
+    }
+
+    private static Stream<Arguments> testDataForCircularDependencyMismatchTest() {
+        return Stream.of(
+                Arguments.of("stringDependsOnBool", "mainBoolFlag", true, "Dog"),
+                Arguments.of("stringDependsOnBool", "mainBoolFlag", false, "Cat"),
+                Arguments.of("stringDependsOnBool", "mainBoolFlag", "1", null),
+                Arguments.of("stringDependsOnBool", "mainBoolFlag", 1, null),
+                Arguments.of("stringDependsOnBool", "mainBoolFlag", 1.0, null),
+                Arguments.of("stringDependsOnString", "mainStringFlag", "private", "Dog"),
+                Arguments.of("stringDependsOnString", "mainStringFlag", "Private", "Cat"),
+                Arguments.of("stringDependsOnString", "mainStringFlag", true, null),
+                Arguments.of("stringDependsOnString", "mainStringFlag", 1, null),
+                Arguments.of("stringDependsOnString", "mainStringFlag", 1.0, null),
+                Arguments.of("stringDependsOnInt", "mainIntFlag", 2, "Dog"),
+                Arguments.of("stringDependsOnInt", "mainIntFlag", 1, "Cat"),
+                Arguments.of("stringDependsOnInt", "mainIntFlag", "2", null),
+                Arguments.of("stringDependsOnInt", "mainIntFlag", true, null),
+                Arguments.of("stringDependsOnInt", "mainIntFlag", 2.0, null),
+                Arguments.of("stringDependsOnDouble", "mainDoubleFlag", 0.1, "Dog"),
+                Arguments.of("stringDependsOnDouble", "mainDoubleFlag", 0.11, "Cat"),
+                Arguments.of("stringDependsOnDouble", "mainDoubleFlag", "0.1", null),
+                Arguments.of("stringDependsOnDouble", "mainDoubleFlag", true, null),
+                Arguments.of("stringDependsOnDouble", "mainDoubleFlag", 1, null)
+        );
+    }
+
+    private static Stream<Arguments> testDataForPrerequisiteFlagOverrideTest() {
+        return Stream.of(
+                Arguments.of("stringDependsOnString", "1", "john@sensitivecompany.com", null, "Dog"),
+                Arguments.of("stringDependsOnString", "1", "john@sensitivecompany.com", OverrideBehaviour.REMOTE_OVER_LOCAL, "Dog"),
+                Arguments.of("stringDependsOnString", "1", "john@sensitivecompany.com", OverrideBehaviour.LOCAL_OVER_REMOTE, "Dog"),
+                Arguments.of("stringDependsOnString", "1", "john@sensitivecompany.com", OverrideBehaviour.LOCAL_ONLY, null),
+                Arguments.of("stringDependsOnString", "2", "john@notsensitivecompany.com", null, "Cat"),
+                Arguments.of("stringDependsOnString", "2", "john@notsensitivecompany.com", OverrideBehaviour.REMOTE_OVER_LOCAL, "Cat"),
+                Arguments.of("stringDependsOnString", "2", "john@notsensitivecompany.com", OverrideBehaviour.LOCAL_OVER_REMOTE, "Dog"),
+                Arguments.of("stringDependsOnString", "2", "john@notsensitivecompany.com", OverrideBehaviour.LOCAL_ONLY, null)
+        );
```

Comment:
Test cases based on `stringDependsOnInt` are missing.

Reply:
The Android allows only map-based simple overrides. The `stringDependsOnInt` test case uses a complex setting override, which is not possible.

Reply:
Are you sure? The JS SDK doesn't allow complex overrides either but according to this, it could be done with simple overrides: https://github.com/configcat/common-js/blob/68e23064e70363614c6f8d8bab9619a67fc32646/test/ConfigV2EvaluationTests.ts#L118-L160